src/Forms/NumericField.php

```diff
@@ -149,28 +178,36 @@ public function Value()
         return $formatter->format($this->value, $this->getNumberType());
     }
 
-    public function setValue($value, $data = null)
+    public function getValueForValidation(): mixed
     {
-        $this->originalValue = $value;
-        $this->value = $this->cast($value);
-        return $this;
+        $value = $this->value;
+        if (is_numeric($value) && is_string($value)) {
+            return $this->cast($value);
+        }
+        return $value;
     }
 
     /**
      * Helper to cast non-localised strings to their native type
-     *
-     * @param string $value
-     * @return float|int
      */
-    protected function cast($value)
+    protected function cast(mixed $value): mixed
     {
-        if (strlen($value ?? '') === 0) {
+        if ($value === false) {
+            return false;
+        }
+        if (is_null($value) || is_string($value) && strlen($value) === 0) {
             return null;
         }
-        if ($this->getScale() === 0) {
-            return (int)$value;
+        // try catch block is used to handle cases where casting the value to string causes an error
+        // e.g. trying to cast an array to a string
+        try {
+            if ($this->getScale() === 0) {
+                return (string) (int) $value;
+            }
+            return (string) (float) $value;
```

Comment:
This feels wrong. What is the specific intention here? Casting to int and then to string seems pointless? Is this try/catch a non-standard conditional that could be replaced with a `is_numeric()` check?

Reply:
$value could be anything, e.g. float, int, numeric string.  There was existing logic to chop of decimal places if $this->getScale() is 0, hence the int cast.  

I've updated the docblock description to say that the point of cast is to get a numeric string

Reply:
> This feels wrong. What is the specific intention here? Casting to int and then to string seems pointless?

This part still applies. I still don't understand what the purpose of this is. Perhaps it just needs an inline code comment explaining it? But you might also need to explain it to me in person so I really get it. It's definitely weird and it's not clear what the intention is.

Reply:
I've retained the logic though refactored the class and added a bunch of inline comments explaining all the things

Reply:
Thank you, that's much clearer - and splitting up the casts makes it look more sensible even if the end result is the same.